contracts/FixedRateRewardsSource.sol

```diff
@@ -0,0 +1,140 @@
+        address _target = rewardsTarget;
+        if (_target != msg.sender) {
+            revert UnauthorizedCaller();
+        }
+
+        // Compute pending rewards
+        rewardAmount = previewRewards();
+
+        // Update timestamp
+        rewardConfig.lastCollect = uint64(block.timestamp);
+
+        if (rewardAmount > 0) {
+            // Should not revert if there's no reward to transfer.
+
+            emit RewardCollected(rewardAmount);
+
+            // Intentionally skipping balance check to save some gas
+            // since `transfer` anyway would fail in case of low balance
+            IERC20(rewardToken).transfer(_target, rewardAmount);
+        }
+    }
+
+    /// @dev Compute pending rewards since last collect
+    /// @return rewardAmount Amount of reward that'll be distributed if collected now
+    function previewRewards() public view returns (uint256) {
+        RewardConfig memory _config = rewardConfig;
+        return (block.timestamp - _config.lastCollect) * _config.rewardsPerSecond;
+        // return _previewRewards(rewardConfig);
+    }
+
+    /// @dev Set address of the strategist
+    /// @param _address Address of the Strategist
+    function setStrategistAddr(address _address) external onlyGovernor {
+        _setStrategistAddr(_address);
+    }
+
+    function _setStrategistAddr(address _address) internal {
+        emit StrategistUpdated(_address);
+        // Can be set to zero to disable
+        strategistAddr = _address;
+    }
+
+    /// @dev Set the address of the contract than can collect rewards
+    /// @param _rewardsTarget contract address that can collect rewards
+    function setRewardsTarget(address _rewardsTarget) external onlyGovernor {
+        _setRewardsTarget(_rewardsTarget);
+    }
+
+    /// @dev Set the address of the contract than can collect rewards
+    /// @param _rewardsTarget contract address that can collect rewards
+    function _setRewardsTarget(address _rewardsTarget) internal {
+        emit RewardsTargetChange(_rewardsTarget, rewardsTarget);
+        // Can be set to zero to disable
+        rewardsTarget = _rewardsTarget;
+    }
+
+    /// @dev Set the rate of reward emission
+    /// @param _rewardsPerSecond Amount of rewardToken to distribute per second
+    function setRewardsPerSecond(uint256 _rewardsPerSecond) external onlyGovernorOrStrategist {
+        _setRewardsPerSecond(_rewardsPerSecond);
+    }
+
+    /// @dev Set the rate of reward emission
+    /// @param _rewardsPerSecond Amount of rewardToken to distribute per second
+    function _setRewardsPerSecond(uint256 _rewardsPerSecond) internal {
+        if (_rewardsPerSecond > type(uint192).max) {
+            revert InvalidRewardRate();
+        }
+
+        // Update storage
+        RewardConfig storage _config = rewardConfig;
+        emit RewardsPerSecondChanged(_rewardsPerSecond, _config.rewardsPerSecond);
+        _config.rewardsPerSecond = uint192(_rewardsPerSecond);
```

Comment:
Updating the rewards per second should likely trigger the collection of rewards, transferring the accumulated rewards up to that point using the `_rewardsPerSecond` value before the update is applied.

Reply:
You could add a non `distributedRewards` variable to keep this for a later distribution.